.github/pull_request_template.md

```diff
@@ -0,0 +1,9 @@
+[Link to the Jira ticket](https://dsdmoj.atlassian.net/browse/EL-XXX)
+
+What I have done on this ticket:
+
+
+Checklist:
+ * all tests pass
+ * changes have full test coverage/repo has 100% coverage
+ * recently rebased on main
```

Comment:
How recent? Do we expect one- or two-day tickets to do this, or do we just trust that if git thinks it can merge it it will be ok? For longer integrations, agree, but I'm not certain I like it in the general checklist

Reply:
is there a reason not to rebase on main before merging? I have always felt it was good practice to do so.

Reply:
I think I've run into some frustrations with it mucking with ongoing conversations on the PR (because they relate to commits, and rebasing and force-pushing replaces the commits) but I guess there's a difference between rebasing before creating your PR and rebasing after making changes based on feedback. I have no reason not to do the former, now I think of it. And I guess as this is the PR _creation_ template I've probably just got the wrong end of the stick.

Also I just wanted to push back slightly and challenge what we think the benefit is, and to clarify what we mean by "recently" so it can be enforced

Reply:
I am going to leave it in for now, perhaps at the next ways of working meeting we can discuss it further, see if we can quantify what 'recently' means so we can enforce it, otherwise we can remove the requirement.